tests/test_io.py

```diff
@@ -116,3 +116,38 @@ def test_normal(self):
         np.testing.assert_array_equal(
             tab.loc[("1993", "pr"), ("JFM",)], self.ds.pr.sel(time="1993", season="JFM")
         )
+
+
+def test_round_bits(datablock_3d):
+    da = datablock_3d(
+        np.random.random((30, 30, 50)),
+        variable="tas",
+        x="lon",
+        x_start=-70,
+        y="lat",
+        y_start=45,
+    )
+    dar = xs.io.round_bits(da, 12)
+    # Close but NOT equal, meaning something happened
+    np.testing.assert_allclose(da, dar, rtol=0.013)
+    assert not (da == dar).any()
```

Comment:
This last check could randomly fail if the `np.random.random` generates numbers with a lots of zeroes in the trailing bits.  Say you have a number in your array whose mantissa bits are
1110101**0000000000000000**
i.e. only the 7 first mantissa bits form a non-trivial chain, then keeping 12 bits will result in the same number. 

This is probably very unlikely to get numbers like this with `np.random.random`

Reply:
If my computations are correct, the probability of obtaining a number with the 12 last bits of mantissa all zeroes is 2^-12, which is small, but possible! I relaxed the test.